static/js/settings_org.js

```diff
@@ -218,7 +218,7 @@ export function extract_property_name($elem, for_realm_default_settings) {
         // ID approach.
         return $elem.attr("name");
     }
-    return /^id_(.*)$/.exec($elem.attr("id").replace(/-/g, "_"))[1];
+    return /^id_(?:authmethod[\da-z]+_)?(.*)$/.exec($elem.attr("id").replace(/-/g, "_"))[1];
```

Comment:
Can you explain why we need the `authmethod` prefix here? 

This is a function used in a lot of places, so we'd like the keep hacks for specific elements to a minimum.

Reply:
I think if we do need a special naming scheme here, it might be better to handle in a separate if statement, like the one above this one; this line is already hard to read with the regular expression, and I think adding this might make it near impossible. Something like "If the ID starts with id_authmethod_, compute the property name by stripping id_authmethod_ from the start".

The other note is that ideally, we don't want to make deep assumptions about what characters appear in the authentication method's name.

Reply:
Thanks for your comments, as I think you've already found out, that prefix was added to anchor the added auth method name in the `id` string, to be able to split it from the property name itself, for the save/discard code.

With the regexp-based approach, I aimed at the least amount of change line-wise, but I will look into the alternative approach suggested.

As for the second one, yeah, I was also thinking about those assumptions and I think a better way (and more defensive programming) would be to use a symbol allowlist-based filtering of the auth method name string to convert it to some known subset of symbols (by simple removal of the prohibited symbol, or replacement with an allowed symbol), before using in the `id` (a rudimentary version of that is the space removal I implemented already). That way we can guarantee the symbol set assumption, which we need to do to avoid putting an unsanitized string into the `id`. Any comments on this approach?

Reply:
All right, I've gone ahead and implemented both ideas/approaches.

Please also let me know if I'm overdoing it with comments, in general I think they never hurt, so I went with "more", based on what I saw in other comments in the code - but I'm still calibrating my code style sense for this fine project, so taking any and all advice 🙂

Reply:
Cool, I think this is pretty reasonable.

We will potentially want to do another pass on making this cleaner down the line as part of a broader refactoring of how we do the data for this setting, but I think this version is good enough to let us merge this.